core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java

```diff
@@ -466,25 +484,29 @@ private ThroughputProperties calculateThroughput(Map<String, String> options) {
 
   @Override
   public boolean namespaceExists(String namespace) throws ExecutionException {
-    return databaseExists(namespace);
+    try {
+      getNamespacesContainer()
+          .readItem(namespace, new PartitionKey(namespace), CosmosNamespace.class);
+      return true;
+    } catch (RuntimeException e) {
+      if (e instanceof CosmosException
+          && ((CosmosException) e).getStatusCode() == CosmosErrorCode.NOT_FOUND.get()) {
+        return false;
+      }
+      throw new ExecutionException("Checking if the namespace exists failed", e);
```

Comment:
Good point, thank you. 
Any checked exception thrown by the `[Cosmos/Cassandra/Dynamoc/Jdbc]Admin` methods will be caught and rethrown by the `CheckedDistributedStorageAdmin` wrapper. All rethrown exception messages will contain the namespace or table name which failed. 
https://github.com/scalar-labs/scalardb/blob/ea4c543465ca430dcbfa0eb627f8036779049351/core/src/main/java/com/scalar/db/common/CheckedDistributedStorageAdmin.java#L202-L208

Considering that, adding the namespace in the error message is not necessary. What do you think?

Reply:
Thanks for checking that. 

> Any checked exception thrown by the [Cosmos/Cassandra/Dynamoc/Jdbc]Admin methods will be caught and rethrown by the CheckedDistributedStorageAdmin wrapper

It sounds okay for now. But `CosmosAdmin#namespaceExists` is a public method and it's not guaranteed that only `CheckedDistributedStorageAdmin.namespaceExists` calls `CosmosAdmin#namespaceExists` even in the future. 

This might be a matter of taste. I'm a bit paranoid about avoiding lack of information and prefer to add information just in case even if some is duplicated in log message. But either is fine with me based on your above check 💪

Reply:
Right, that's a sensible concern. 
Almost all methods of the Admins are concerned with this, so I will revise everything in a separate PR.